Code/Framework/AzFramework/AzFramework/Network/IRemoteTools.h

```diff
@@ -0,0 +1,262 @@
+#include <AzCore/std/containers/deque.h>
+#include <AzCore/std/containers/unordered_map.h>
+#include <AzCore/std/smart_ptr/intrusive_ptr.h>
+#include <AzCore/std/string/string.h>
+
+namespace AZ
+{
+    template<typename T>
+    class Interface;
+}
+
+namespace AzFramework
+{
+    class RemoteToolsMessage
+    {
+    public:
+        AZ_CLASS_ALLOCATOR(RemoteToolsMessage, AZ::OSAllocator, 0);
+        AZ_RTTI(RemoteToolsMessage, "{8512328C-949D-4F0C-B48D-77C26C207443}");
+
+        RemoteToolsMessage()
+            : m_msgId(0)
+            , m_senderTargetId(0)
+            , m_customBlob(nullptr)
+            , m_customBlobSize(0)
+            , m_refCount(0)
+            , m_isBlobOwner(false)
+            , m_immediateSelfDispatch(false)
+        {
+        }
+        RemoteToolsMessage(AZ::u64 msgId)
+            : m_msgId(msgId)
+            , m_senderTargetId(0)
+            , m_customBlob(nullptr)
+            , m_customBlobSize(0)
+            , m_refCount(0)
+            , m_isBlobOwner(false)
+            , m_immediateSelfDispatch(false)
+        {
+        }
+
+        virtual ~RemoteToolsMessage();
+
+        //! Add a custom data blob to a Remote Tools message
+        //! @param blob Pointer to the start of the data blob
+        //! @param blobSize Size of the data blob
+        //! @param ownBlob Whether this message owns the life cycle of the blob memory
+        void AddCustomBlob(const void* blob, size_t blobSize, bool ownBlob = false);
+
+        //! Sets if this message can be immediately dispatched if the receipient is itself
+        //! @param immediateSelfDispatchEnabled Whether this message can be immediately dispatched
+        void SetImmediateSelfDispatchEnabled(bool immediateSelfDispatchEnabled);
+
+        //! Gets if this message can be immediately dispatched if the receipient is itself
+        //! @return Whether this message can be immediately dispatched
+        bool IsImmediateSelfDispatchEnabled() const;
+
+        //! Gets the custom data blob assosciated with this message
+        //! @return Memory address of the start of the blob or nullptr if there is none
+        const void* GetCustomBlob() const;
+
+        //! Gets the size of the custom data blob associated with this message
+        //! @return Size of the data blob or 0 if there is none
+        size_t GetCustomBlobSize() const;
+
+        //! Gets the numeric ID of this message
+        //! @return The ID of this message
+        AZ::u64 GetId() const;
+
+        //! Gets this message's sender's ID
+        //! @return ID of the message's sender
+        AZ::u32 GetSenderTargetId() const;
+
+        //! Sets the ID of this message's sender
+        //! @param senderTargetId The ID to set for this message's sender
+        void SetSenderTargetId(AZ::u32 senderTargetId);
+
+    protected:
+        AZ::u64 m_msgId;
+        AZ::u32 m_senderTargetId;
+        const void* m_customBlob;
+        AZ::u32 m_customBlobSize;
```

Comment:
You could encapsulate the custom blob in an `AZStd::span<AZStd::byte const>`

Reply:
I'll dig into this, thanks!